enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,441 @@
+
+The OpenStack cluster might make the baremetal nodes accessible to the outside
+world, but exposing them via floating IPs might represent an increase in
+complexity and is currently not an explicit goal.
+
+## Proposal
+
+This enhancement is less about adding a new logical piece, than it is about
+uncovering unknowns in a configuration that might work out-of-the-box in
+selected environments.
+
+Therefore, the implementation consists of:
+* identifying a reference architecture
+* anticipating probable issues
+* testing and fixing errors
+
+### Characteristics of the reference architecture
+
+Prerequisites:
+
+* The OpenStack cloud runs the [Bare Metal (ironic)][openstack-ironic] service
+  to provide baremetal management
+* The OpenStack [Compute (nova)][openstack-nova] and [Networking
+  (neutron)][openstack-neutron] services are configured to serve bare metal
+  nodes using the Nova API
+* All of the additional complexity inherent to baremetal is handled by the IaaS
+  cloud and its administrators:
+    * Enrolling new hardware and discovering its components
+    * Wiping the disks between deployments
+    * Booting the system up from the desired image
+    * Networking
+* The choice between deploying a virtual or baremetal machine is done by
+  selecting an appropriate Nova flavor and Neutron network
+* Both virtual and baremetal machines consume the same image from glance
+* The installer will have no insight into the Ironic networking
+    * All it will do is create a node with the user-specified flavor and
+      network
+
+We target four distinct use-cases:
+
+1. Install with BM workers on the installer-provisioned network
+2. Install with BM workers on a preexisting network
+3. Install with both VM and BM workers (BMs on a preexisting network)
+4. Install on VMs then add BM workers (BMs on a preexisting network)
+
+### Implementation work
+
+#### Case 1. Install with BM workers on the installer-provisioned network
+
+Booting BMs in a virtual network requires specific hardware; this case is a
+simplified one and is not expected to be the most represented in real-world
+installations.
+
+Prerequisites for the user:
+
+* Being able to create BMs on a virtual subnet
+
+Expected work:
+
+* Document how to use:
+  * set the Ironic flavor in `install-config.yaml`
+* Adapt the wait-for install-complete timeout, either in code or in
+  documentation, in order to accomodate for the BM booting times.
+
+#### Case 2. Install with BM workers on a preexisting network
+
+This case leverages [Provider Networks and Custom
+Subnets](openstack-customer-provided-subnets.md). The cluster is installed on
+the same pre-existing network where the BMs boot.
+
+Prerequisites for the user:
+
+* Being able to create VMs on the preexisting subnet
+
+Expected work:
+
+* Document how to use:
+  * set the appropriate Ironic-backed Nova flavor in `install-config.yaml`
+  * set the workers set to run on a custom network in `install-config.yaml`
```

Comment:
Thanks for pointing this out; it's a leftover from a previous draft.
In this "case 2: install with BM workers on a preexisting network" we would just use BYON to install the whole cluster on the Provider network.

By the way, this is one of my main uncertainties right now: I am not sure that spinning VMs is at all possible on the BM subnet, and I am not sure it is a valid use-case to target anyway. If not, then it will be necessary to deploy the Control plane in the tenant network and make sure it talks with the BM subnet.